src/main/proto/ga4gh/read_service.proto

```diff
@@ -91,6 +103,52 @@ message GetReadGroupSetRequest {
   string read_group_set_id = 1;
 }
 
+// ******************  /readgroups  *********************/
+
+// This request maps to the body of `POST /readgroups/search` as JSON.
+
+
+message SearchReadGroupsRequest {
+  // The dataset to search.
```

Comment:
Something needs to be required - I'd say pick one or the other between dataset and read group set ID, make it required, and drop the other.

Searching across the entire repository without a parent ID is not going to be feasible for all implementations and is not consistent with the other endpoints.